pkg/cluster/rook_ceph.go

```diff
@@ -196,81 +227,102 @@ func (c *Controller) SetDeviceHealthMetricsReplication(rookVersion semver.Versio
 		minSize = 2
 	}
 
-	c.Log.Debugf("Ensuring %s replication level is %d", CephDeviceHealthMetricsPool, level)
+	poolName := CephDeviceHealthMetricsPool
+	if cephVersion != "" {
+		cephVersionSemver, err := semver.Parse(cephVersion)
+		if err != nil {
+			return false, errors.Wrapf(err, "parse ceph version %q", cephVersion)
+		}
+		if cephVersionSemver.Major >= 17 {
+			poolName = CephDeviceHealthMetricsPoolQuincy
+		}
+	}
+
+	c.Log.Debugf("Ensuring %s replication level is %d", poolName, level)
 
-	err := c.cephOSDPoolSetSize(rookVersion, CephDeviceHealthMetricsPool, level, cephcluster)
+	err := c.cephOSDPoolSetSize(rookVersion, poolName, level, cephcluster)
 	if err != nil {
-		return false, errors.Wrapf(err, "scale %s pool size", CephDeviceHealthMetricsPool)
+		return false, errors.Wrapf(err, "scale %s pool size", poolName)
 	}
-	err = c.cephOSDPoolSetMinSize(rookVersion, CephDeviceHealthMetricsPool, minSize)
+	err = c.cephOSDPoolSetMinSize(rookVersion, poolName, minSize)
 	if err != nil {
-		return false, errors.Wrapf(err, "scale %s pool min_size", CephDeviceHealthMetricsPool)
+		return false, errors.Wrapf(err, "scale %s pool min_size", poolName)
 	}
 
 	return true, nil
 }
 
-func (c *Controller) ReconcileMonCount(ctx context.Context, count int) error {
+func (c *Controller) ReconcileMonCount(ctx context.Context, nodeCount int) error {
 	// single mon for 1 or 2 node cluster, 3 mons for all other clusters
-	if count < 3 {
-		count = 1
-	} else {
-		count = 3
+	monCount := 3
+	if nodeCount < 3 {
+		monCount = 1
```

Comment:
Do we really need it (it seems make harder read the code) could we not to do the check directly since the values are fixed like:
```
if (cluster.Spec.Mon.Count == 3) || (cluster.Spec.Mon.Count < 3) {
       c.Log.Debugf("Will not reduce mon count from %s to %s", cluster.Spec.Mon.Count, monCount)
      return nil
}
```

Reply:
This is not equivalent because we never set mon count to 2